src/unnamed_rcv_thing/cvr_loader.py

```diff
@@ -0,0 +1,81 @@
+import pandas as pd
+import pathlib
+import os
+from pydantic import BaseModel
+from typing import Callable
+from unnamed_rcv_thing.profile import PreferenceProfile
+from unnamed_rcv_thing.ballot import Ballot
+from pandas.errors import EmptyDataError, DataError
+
+class CVRLoader(BaseModel):
+    """
+    load_func (Callable([str, PreferenceProfile])):
+      given a file path, loads cvr into a preference profile
+    """
+    load_func: Callable[[str], PreferenceProfile]
+
+    class Config:
+        allow_mutation = False
+    
+    def load_cvr(self, fpath: str, id_col: int = None) -> PreferenceProfile:
+        """
+        checks for the following errors and calls load_func to parse cvr file
+        Args:
+            fpath (str): path to cvr file
+            id_col (int, optional): index for the column with voter ids. Defaults to None.
+
+        Raises:
+            FileNotFoundError: if fpath is invalid
+            EmptyDataError: if dataset is empty
+            ValueError: if the voter id column has missing values
+            DataError: if the voter id column has duplicate values
+
+        Returns:
+            PreferenceProfile: preference schedule that contains all the parsed ballots
+        """
+        if not os.path.isfile(fpath):
+            raise FileNotFoundError(f'File with path {fpath} cannot be found')
+        cvr_path = pathlib.Path(fpath)
+        df = pd.read_csv(cvr_path, on_bad_lines='error', encoding="utf8")
```

Comment:
A question I have about this implementation is whether or not the load_func should be wrapped like this.

One option would be to move this logic into rank_column_csv & not assume that all files being loaded are CSVs.

This would mean there's no need for CVRLoader, load functions would be called directly.